dns-policy-operator/src/models.py

```diff
@@ -0,0 +1,112 @@
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Common data structures used in this charm."""
+
+import hashlib
+
+import pydantic
+from charms.bind.v0.dns_record import RecordClass, RecordType, RequirerEntry
+
+
+class DnsEntry(pydantic.BaseModel):
+    """Class used to represent a DNS entry.
+
+    Attributes:
+        domain: example: "dns.test"
+        host_label: example: "admin"
+        ttl: example: 600
+        record_class: example: "IN"
+        record_type: example: "A"
+        record_data: example: "42.42.42.42"
+    """
+
+    domain: str = pydantic.Field(min_length=1)
+    host_label: str = pydantic.Field(min_length=1)
+    ttl: int
```

Comment:
I think a range might be in order here? 1 to 604800?

Reply:
Added